src/lib/structures/managers/StickyRoleManager.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @typescript-eslint/explicit-member-accessibility */
+import { GuildSettings, StickyRole } from '@lib/types/settings/GuildSettings';
+import { Guild } from 'discord.js';
+
+export class StickyRoleManager {
+
+	#guild: Guild;
+
+	public constructor(guild: Guild) {
+		this.#guild = guild;
+	}
+
+	private get entries() {
+		return this.#guild.settings.get(GuildSettings.StickyRoles);
+	}
+
+	public get(userID: string): readonly string[] {
+		return this.entries.find(entry => entry.user === userID)?.roles ?? [];
+	}
+
+	public async fetch(userID: string, extraContext?: unknown): Promise<readonly string[]> {
```

Comment:
extraContent should be typed like this because we're always passing in the author (and if we're not then we should)

```ts
type ExtraContent<T> = T & { author: string }
```